### ISSUE TITLE
[YSQL] Inserting NULL into column with UNIQUE INDEX causes crash

### ISSUE THREAD

**Issue by frozenspider** [YugaByte/yugabyte-db#1058]
psql:
```sql
postgres=# CREATE TABLE k.num2 (pk int PRIMARY KEY, i int, vi bigint);
CREATE TABLE
postgres=# CREATE UNIQUE INDEX k_num2_vi ON k.num2 (vi);
CREATE INDEX
postgres=# INSERT INTO k.num2 (pk, i, vi) VALUES (0, 0, NULL);
ERROR:  Timed out: Write(tablet: 9f71927d692747ce9e9eb492d13cb394, num_ops: 1, num_attempts: 343, txn: 85932c01-2bff-4d35-829c-2cf1beed405f) passed its deadline 376402.720s (now: 376402.794s): Network error (yb/util/net/socket.cc:570): Recv() got EOF from remote (error 58)
```
Server:
```
F0322 19:03:43.199790 44023808 enums.h:176] Invalid value of enum ValueType (full enum type: yb::docdb::ValueType, expression: type_): 88.
Fatal failure details written to /Users/fs/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2019-03-22T19_03_43.pid28222.txt
F20190322 19:03:43 ../../src/yb/util/enums.h:176] Invalid value of enum ValueType (full enum type: yb::docdb::ValueType, expression: type_): 88.
    @        0x105ca8500  google::LogDestination::LogToSinks()
    @        0x105ca750b  google::LogMessage::SendToLog()
    @        0x105ca7eb5  google::LogMessage::Flush()
    @        0x105cac9bf  google::LogMessageFatal::~LogMessageFatal()
    @        0x105ca8ea9  google::LogMessageFatal::~LogMessageFatal()
    @        0x103f1b70b  yb::FatalInvalidEnumValueInternal<>()
    @        0x103f1bd62  yb::docdb::PrimitiveValue::AppendToKey()
    @        0x103ebca0b  yb::docdb::DocKey::AppendTo()
    @        0x103ebcc30  yb::docdb::DocKey::EncodeAsRefCntPrefix()
    @        0x103f0fd74  yb::docdb::PgsqlWriteOperation::Init()
    @        0x103a71d22  yb::tablet::Tablet::KeyValueBatchFromPgsqlWriteBatch()
    @        0x103a721c4  yb::tablet::Tablet::AcquireLocksAndPerformDocOperations()
    @        0x103aad8c0  yb::tablet::TabletPeer::WriteAsync()
    @        0x1039056f8  yb::tserver::TabletServiceImpl::Write()
    @        0x1049603e7  yb::tserver::TabletServerServiceIf::Handle()
    @        0x105170120  yb::rpc::ServicePoolImpl::Handle()
    @        0x105170c85  yb::rpc::TasksPool<>::WrappedTask::Run()
    @        0x1051788e7  yb::rpc::(anonymous namespace)::Worker::Execute()
    @        0x105809c2f  yb::Thread::SuperviseThread()
    @     0x7fff53850661  _pthread_body
    @     0x7fff5385050d  _pthread_start
    @     0x7fff5384fbf9  thread_start

*** Check failure stack trace: ***
    @        0x105ca77b2  google::LogMessage::SendToLog()
    @        0x105ca7eb5  google::LogMessage::Flush()
    @        0x105cac9bf  google::LogMessageFatal::~LogMessageFatal()
    @        0x105ca8ea9  google::LogMessageFatal::~LogMessageFatal()
    @        0x103f1b70b  yb::FatalInvalidEnumValueInternal<>()
    @        0x103f1bd62  yb::docdb::PrimitiveValue::AppendToKey()
    @        0x103ebca0b  yb::docdb::DocKey::AppendTo()
    @        0x103ebcc30  yb::docdb::DocKey::EncodeAsRefCntPrefix()
    @        0x103f0fd74  yb::docdb::PgsqlWriteOperation::Init()
    @        0x103a71d22  yb::tablet::Tablet::KeyValueBatchFromPgsqlWriteBatch()
    @        0x103a721c4  yb::tablet::Tablet::AcquireLocksAndPerformDocOperations()
    @        0x103aad8c0  yb::tablet::TabletPeer::WriteAsync()
    @        0x1039056f8  yb::tserver::TabletServiceImpl::Write()
    @        0x1049603e7  yb::tserver::TabletServerServiceIf::Handle()
    @        0x105170120  yb::rpc::ServicePoolImpl::Handle()
    @        0x105170c85  yb::rpc::TasksPool<>::WrappedTask::Run()
    @        0x1051788e7  yb::rpc::(anonymous namespace)::Worker::Execute()
    @        0x105809c2f  yb::Thread::SuperviseThread()
    @     0x7fff53850661  _pthread_body
    @     0x7fff5385050d  _pthread_start
    @     0x7fff5384fbf9  thread_start

Process finished with exit code 6
```

Note that PostgreSQL allows duplicate NULL values for a column with unique index: https://www.postgresql.org/docs/11/ddl-constraints.html#DDL-CONSTRAINTS-UNIQUE-CONSTRAINTS

**Comment by frozenspider** [comment]
Note that this is NOT reproducible with YCQL:
```cql
cqlsh> CREATE TABLE k.num2 (pk int PRIMARY KEY, i int, vi bigint) WITH transactions = {'enabled': true};
cqlsh> INSERT INTO k.num2 (pk, i, vi) VALUES (0, 0, NULL);
cqlsh> TRUNCATE TABLE k.num2;
cqlsh> CREATE UNIQUE INDEX k_num2_vi ON k.num2 (vi);
cqlsh> INSERT INTO k.num2 (pk, i, vi) VALUES (0, 0, NULL);
cqlsh> INSERT INTO k.num2 (pk, i, vi) VALUES (1, 0, NULL);
InvalidRequest: Error from server: code=2200 [Invalid query] message="Execution Error. Duplicate value disallowed by unique index k_num2_vi
INSERT INTO k.num2 (pk, i, vi) VALUES (1, 0, NULL);
       ^^^^
 (error -300)"
```

**Comment by frozenspider** [comment]
This was partially fixes as a part of #945 - `NULL`s are now inserted normally but aren't exempt from unique constraints (as they should be per SQL standard)
